Downgrade -O2 down to -O1 on UNREG arches to speed builds up
and decrease amount of generated C code.
diff --git a/mk/config.mk.in b/mk/config.mk.in
index 189439ec7c..c6179fa03e 100644
--- a/mk/config.mk.in
+++ b/mk/config.mk.in
@@ -901,6 +901,15 @@ GhcStage2HcOpts=
 GhcStage3HcOpts=
 
 GhcLibHcOpts=
+
+# clamp down to -O1
+libraries/binary_EXTRA_HC_OPTS     += -O1
+libraries/containers_EXTRA_HC_OPTS += -O1
+libraries/bytestring_EXTRA_HC_OPTS += -O1
+libraries/xhtml_EXTRA_HC_OPTS      += -O1
+utils/haddock_EXTRA_HC_OPTS        += -O1
+
+GhcRtsHcOpts=
 endif
 
 # Note [Disable -O2 in unregisterised mode]
